gdk4/src/clipboard.rs

```diff
@@ -65,25 +65,22 @@ impl Clipboard {
     }
 
     #[allow(clippy::type_complexity)]
-    pub fn read_future(
+    pub async fn read_future(
         &self,
         mime_types: &[&str],
         io_priority: glib::Priority,
-    ) -> Pin<
-        Box<
-            dyn future::Future<Output = Result<(gio::InputStream, GString), glib::Error>> + 'static,
-        >,
-    > {
+    ) -> Result<(gio::InputStream, GString), glib::Error> {
         let mime_types = mime_types
             .iter()
             .copied()
             .map(String::from)
             .collect::<Vec<_>>();
```

Comment:
This is an API change now btw. The previous version was only borrowing `mime_types` for the duration of the call, this one is borrowing it until the resulting future is gone.

Not necessarily a problem but something to be aware of.